### PR TITLE
UINV-498: display only active account numbers in dropdown list

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UINV-498](https://issues.folio.org/browse/UINV-498) - display only active account numbers in dropdown list
## Approach

https://github.com/folio-org/ui-invoice/assets/116072773/9c03fe55-de97-4e0b-b600-7f821865dae8

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/invoices/utils.js [hunk 1]
```diff
@@ -118,3 +119,10 @@ export const getCommonInvoiceLinesFormatter = (currency, ordersMap, orderlinesMa
     line.referenceNumbers?.map(({ refNumber }) => refNumber)?.join(', ') || <NoValue />
   ),
 });
+
+export const getActiveAccountNumbers = (accounts = []) => {
+  console.log('accounts', accounts);
```
Comment (NikitaSedyx) [inline src/invoices/utils.js:124]: o_O

**Comment by usavkov-epam** [comment]
Is it expected that after changing an already assigned account to inactive status, it displays an empty value in the edit form?

https://github.com/folio-org/ui-invoice/assets/88109087/c36622a9-4625-49ff-8a2c-082c177a581c

**Review by BogdanDenis — APPROVED** [review]

**Review by mariia-aloshyna — APPROVED** [review]

File: src/invoices/InvoiceLineForm/InvoiceLineForm.js [hunk 2]
```diff
@@ -43,21 +44,18 @@ import {
...
 } from '../../common/utils';
 import {
   convertToInvoiceLineFields,
+  getActiveAccountNumbers,
 } from '../utils';
 import AdjustmentsForm from '../AdjustmentsForm';
-import {
-  SECTIONS_INVOICE_LINE_FORM as SECTIONS,
-} from '../constants';
+import { ACCOUNT_STATUS, SECTIONS_INVOICE_LINE_FORM as SECTIONS } from '../constants';
```
Comment (usavkov-epam) [inline src/invoices/InvoiceLineForm/InvoiceLineForm.js:58]: could you split it into several lines, please

File: src/invoices/InvoiceLineForm/InvoiceLineForm.js [hunk 3]
```diff
@@ -123,10 +121,12 @@ const InvoiceLineForm = ({
   } = initialValues;
   const totalAmount = calculateTotalAmount(formValues, invoice.currency);
   const isEditPostApproval = IS_EDIT_POST_APPROVAL(id, invoiceLineStatus);
+  const initialAccountNumber = useRef(accountNumber);
```
Comment (usavkov-epam) [inline src/invoices/InvoiceLineForm/InvoiceLineForm.js:124]: Why do you need `useRef` to get the initial value of the account number? Form initial values should not change during form re-renders.

File: src/invoices/InvoiceLineForm/InvoiceLineForm.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ import { useHistory } from 'react-router';
 import {
   find,
   get,
-  uniq,
+  values,
```
Comment (usavkov-epam) [inline src/invoices/InvoiceLineForm/InvoiceLineForm.js:10]: is there any usage of that?

**Comment by usavkov-epam** [comment]
Please fix linter errors and warnings on the new code.

**Comment by usavkov-epam** [comment]
The warning banner is displayed for "empty" value.


https://github.com/folio-org/ui-invoice/assets/88109087/a8be0233-94c6-45d4-823c-c8aa7c239fd2

**Comment by usavkov-epam** [comment]
And there are still a bunch of lint errors

**Comment by alisher-epam** [comment]
> The warning banner is displayed for "empty" value.
> 
>  chrome_XQvQ6XPgYs.mp4

Fixed this one.

**Review by usavkov-epam — APPROVED** [review]

**Pull request merged**